### PR TITLE
feat: legend table is implemeted using tanstack table

### PR DESCRIPTION
**Pull request by Chandru-HM**
## Overview
This PR is for ticket #2277 and implemented legend table using tanstack table library
**Note**: need to set SHOW_LEGEND_TANSTACK_TABLE  to true in local storage to see legend table using tanstack

## Verifying Changes
[tanstack-table.webm](https://github.com/awslabs/iot-app-kit/assets/142866907/e2477860-e35c-46f6-90f5-322be5c72753)

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/components/chart/legend/sotingIcons.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (mnischay) [inline packages/react-components/src/components/chart/legend/sotingIcons.tsx:1]: can we store these as .svgs ?

File: packages/react-components/src/components/chart/legend/useChartsLegend.tsx [hunk 2]
```diff
@@ -22,6 +22,13 @@ import { LEGEND_NAME_MIN_WIDTH_FACTOR } from '../eChartsConstants';
 import Hide from './hide.svg';
```
Comment (mnischay) [inline packages/react-components/src/components/chart/legend/useChartsLegend.tsx:22]: move the tanstack logic to a completely new file, so its clean and devoid of the cloudscape logic

**Review by mnischay — CHANGES_REQUESTED** [review]

File: packages/react-components/src/components/chart/legend/useChartsLegend.tsx [hunk 3]
```diff
@@ -201,6 +227,20 @@ const useChartsLegend = ({
     });
     setColumnDefinitions([legendColumnDefinition, ...tcColumnDefinitions]);
 
+    const tcColumnDefinitionsTanstack = graphic.map((g) => {
```
Comment (mnischay) [inline packages/react-components/src/components/chart/legend/useChartsLegend.tsx:230]: same with this... everything w.r.t tanstack should its own appropriate files

File: packages/react-components/src/components/chart/legend/tanstackTable.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import { sortingIconAsc, sortingIconDesc } from './sotingIcons';
+
+type tanstackTableOptions = {
+  data: object[];
+  columnDefinitions: ColumnDef<object, string>[];
+  stickyColumns?: { first: number };
+  stickyHeader?: boolean;
+};
+
+export const TanstackTable = ({ data, columnDefinitions, stickyColumns, stickyHeader }: tanstackTableOptions) => {
```
Comment (tracy-french) [inline packages/react-components/src/components/chart/legend/tanstackTable.tsx:32]: Overall implementation of <TanstackTable /> component looks good. @Chandru-HM how was the implementation for you? Straight-forward or did you run into any issues?

File: packages/react-components/src/components/chart/legend/sotingIcons.tsx (hunk 1 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/sotingIcons.tsx]: updated.

File: packages/react-components/src/components/chart/legend/useChartsLegend.tsx (hunk 2 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/useChartsLegend.tsx]: done

File: packages/react-components/src/components/chart/legend/useChartsLegend.tsx (hunk 3 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/useChartsLegend.tsx]: updated

File: packages/react-components/src/components/chart/legend/tanstackTable.tsx (hunk 4 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/tanstackTable.tsx:34]: Thank you.
It's good, more customizable and Straight-forward. but for few I made custom, due to no default property is available (like sticky header, sticky column and sorting)

File: packages/react-components/src/components/chart/legend/useChartsLegendTanstack.tsx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        ref={nameRef}
+        title={isNameTruncated ? name : undefined}
+      >
+        {name}
+      </div>
+    </div>
+  );
+};
+
+const useChartsLegendTanstack = ({
```
Comment (mnischay) [inline packages/react-components/src/components/chart/legend/useChartsLegendTanstack.tsx:135]: may be worth adding an exit condition like 

if(!enabed) return { columnDefinitions: [], items: []}

Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/useChartsLegendTanstack.tsx]: updated.

File: packages/react-components/src/components/chart/legend/tanstackLegend.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { DataStream } from '@iot-app-kit/core';
+import { ColumnDef } from '@tanstack/react-table';
+
+import { SeriesOption } from 'echarts';
+import useChartsLegendTanstack from './useChartsLegendTanstack';
+
+import { InternalGraphicComponentGroupOption } from '../trendCursor/types';
+import { TanstackTable } from './tanstackTable';
+import './legend.css';
```
Comment (mnischay) [inline packages/react-components/src/components/chart/legend/tanstackLegend.tsx:10]: nit: do you need this CSS file imported ?

**Review by mnischay — DISMISSED** [review]

File: packages/react-components/src/components/chart/legend/tanstackTable.css [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+.tanstack-table-sticky-column-body {
+  left: 0;
+  position: sticky;
+  z-index: 1;
+  white-space: nowrap;
+  cursor: default;
+}
+
+.tanstack-table-sorting-icon {
```
[the remark this answers is not in the document]
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/tanstackTable.css]: updated

File: packages/react-components/src/components/chart/legend/tanstackLegend.tsx (hunk 6 above)
Comment (Chandru-HM) [reply packages/react-components/src/components/chart/legend/tanstackLegend.tsx]: updated

**Review by mnischay — APPROVED** [review]

**Review by corteggiano — APPROVED** [review]

**Pull request merged**